tools/gulp/tasks/screenshots.ts

```diff
@@ -45,14 +45,46 @@ task('screenshots', () => {
     const firebaseApp = connectFirebaseScreenshots();
     const database = firebaseApp.database();
 
+    // If this task hasn't completed in 8 minutes, close the firebase connection.
+    const timeoutId = setTimeout(() => {
+      console.error('Screenshot tests did not finish in 8 minutes, closing Firebase connection.');
+      return firebaseApp.delete();
+    }, 60 * 1000 * 8);
+
+    let lastActionTime = Date.now();
     return uploadTravisJobInfo(database, prNumber)
-      .then(() => downloadGoldScreenshotFiles(database))
-      .then(() => compareScreenshotFiles(database, prNumber))
-      .then(passedAll => setPullRequestResult(database, prNumber, passedAll))
-      .then(() => uploadScreenshotsData(database, 'diff', prNumber))
-      .then(() => uploadScreenshotsData(database, 'test', prNumber))
-      .catch((err: any) => console.error(err))
-      .then(() => firebaseApp.delete());
+      .then(() => {
+        console.log(`  Downloading screenshot golds from Firebase...`);
+        lastActionTime = Date.now();
+        return downloadGoldScreenshotFiles(database);
+      })
+      .then(() => {
+        console.log(`  Downloading golds done (took ${Date.now() - lastActionTime}ms)`);
```

Comment:
If you want these to be formatted better, you can `import {log} from 'gulp-util'`.